In [1]:
import pandas as pd
import altair as alt

In [6]:
resenas = pd.read_csv('../results/all_amazon_reviews.csv')

In [60]:
resenas['star_rating'].value_counts()

star_rating
5.0 de 5 estrellas    203
No rating              95
4.0 de 5 estrellas     47
1.0 de 5 estrellas     22
2.0 de 5 estrellas      7
3.0 de 5 estrellas      6
Name: count, dtype: int64

In [58]:
base = alt.Chart(resenas).mark_bar().encode(alt.X("star_rating", title ='Estrellas'), 
    alt.Y('count()', title = 'Conteo'), alt.Color("star_rating:N"))

# barras de la gráfica
bar = base.mark_bar().properties(width=500,height=300,
    title='Conteo de calificacion por cantidad de estrellas')
# texto de la gráfica
text = base.mark_text(baseline='bottom').encode(
    alt.Text('count()', format=",.0f"))

# poner barras y texto de la gráfica juntos
bar + text

alt.LayerChart(...)

In [66]:
#Creamos un DF con los Value_count y agregamos una columna con el porcentaje que representa cada conteo
resenas_count = resenas['star_rating'].value_counts().reset_index()
resenas_count.columns = ['star_rating','Count']
resenas_count['Porcentaje'] = (resenas_count['Count'] / resenas_count['Count'].sum()) * 100
resenas_count

star_rating  Count  Porcentaje
0  5.0 de 5 estrellas    203   53.421053
1           No rating     95   25.000000
2  4.0 de 5 estrellas     47   12.368421
3  1.0 de 5 estrellas     22    5.789474
4  2.0 de 5 estrellas      7    1.842105
5  3.0 de 5 estrellas      6    1.578947

#### De la gráfica y de este análisis podemos ver que existen 53 % de reseñas que dan excelente calificación con el máximo de 5 estrellas.
#### Un 12% da muy buena calificación con 4 estrellas. 
#### Un 6 % da una calificación Media con 3 estrellas
#### Un 3% da calificación Mala y Muy  mala con 1 y 2 estrellas.
#### Y un 25% no puso calificación con estrellas en su reseña lo cual no es posible saber en base a estrellas si su reseña es buena o mala y puede caer en cualquier categoría. 